KAFKA-8155: Add 2.2.0 release to system tests

mjsax opened the pull request:
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

mjsax committed:
KAFKA-8155: Add 2.2.0 release to system tests

mjsax commented:
Triggered _all_ system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2515/

Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna

mjsax commented:
Should this be cherry-picked to `2.2` branch? Not sure.

bbejeck reviewed (approved):
LGTM

@mjsa The `tests/docker/Dockerfile` will need to be updated as well, I'm sure you know just putting a reminder here.

bbejeck commented:
Java 8 failed with `org.apache.kafka.streams.integration.MetricsIntegrationTest.testStreamMetric`
Java 11 failed with `kafka.admin.ResetConsumerGroupOffsetTest.testResetOffsetsNotExistingGroup`

retest this please

mjsax commented:
> @mjsa The tests/docker/Dockerfile will need to be updated as well, I'm sure you know just putting a reminder here.

Thanks Bill. That would have slipped...

mjsax commented:
Retriggered system test that where doomed to fail: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2519/

mjsax commented:
Specified wrong branch for system test. Retriggered: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2526/

mjsax commented:
Rebased to resolve merge conflict. Fixed artifact issue. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2591/

mjsax commented:
Rebased this to get latest system test fixes. New system test run: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2631/

abbccdda commented on streams/upgrade-system-tests-22/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.Properties;
+
+public class StreamsUpgradeTest {
+
```
nit: remove extra line

abbccdda commented on streams/upgrade-system-tests-22/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:
```diff
@@ -0,0 +1,91 @@
+
+            @Override
+            public void init(final ProcessorContext context) {
+                System.out.println("[2.1] initializing processor: topic=data taskId=" + context.taskId());
```
Should this be 2.2?

mjsax replied:
Good catch!

abbccdda reviewed (commented):
Thanks for the hard work! Just some minor comments

cadonna commented on gradle/dependencies.gradle:
```diff
@@ -79,6 +79,7 @@ versions += [
   kafka_11: "1.1.1",
   kafka_20: "2.0.1",
   kafka_21: "2.1.0",
```
Should that be `2.1.1`?
I am asking because this list enumerates only the most recent bugfix releases and you removed `2.1.0` in `tests/docker/Dockerfile`.

mjsax replied:
This is updated in #6596 -- separate PR to allow cherry-picking to older branches.

cadonna commented on tests/docker/Dockerfile:
```diff
@@ -66,6 +65,7 @@ RUN curl -s "$KAFKA_MIRROR/kafka-streams-1.0.2-test.jar" -o /opt/kafka-1.0.2/lib
 RUN curl -s "$KAFKA_MIRROR/kafka-streams-1.1.1-test.jar" -o /opt/kafka-1.1.1/libs/kafka-streams-1.1.1-test.jar
 RUN curl -s "$KAFKA_MIRROR/kafka-streams-2.0.1-test.jar" -o /opt/kafka-2.0.1/libs/kafka-streams-2.0.1-test.jar
 RUN curl -s "$KAFKA_MIRROR/kafka-streams-2.1.0-test.jar" -o /opt/kafka-2.1.0/libs/kafka-streams-2.1.0-test.jar
```
Should this be `kafka-streams-2.1.1-test.jar`?

mjsax replied:
As above.

mjsax committed:
Github comments

mjsax commented:
Updates this.

Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2655/

mjsax commented:
Rebased to resolve merge conflict. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2666/

mjsax committed:
fix

mjsax commented:
Fix failing system test. Retriggered build: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2679/

guozhangwang commented:
LGTM. Please feel free to merge afterwards.

abbccdda reviewed (approved):
Great work!

mjsax commented:
System test passed. Jenkins passed. Merging this now.

mjsax commented:
Merged to `trunk` and cherry-picked to `2.3` branch.
